tests/test_codegen_guided_aws.py

```diff
@@ -0,0 +1,239 @@
+from rpdk.core.project import Project
+from rpdk.java.codegen import JavaArchiveNotFoundError, JavaLanguagePlugin
+
+RESOURCE = "DZQWCC"
+
+
+@pytest.fixture
+def project(tmpdir):
+    project = Project(root=tmpdir)
+    mock_cli = MagicMock(side_effect=mock_input_with_validation)
+    with patch.dict(
+        "rpdk.core.plugin_registry.PLUGIN_REGISTRY",
+        {"test": lambda: JavaLanguagePlugin},
+        clear=True,
+    ), patch("rpdk.java.codegen.input_with_validation", new=mock_cli):
+        project.init("AWS::Foo::{}".format(RESOURCE), "test")
+    return project
+
+
+def mock_input_with_validation(prompt, validate):  # pylint: disable=unused-argument
+    if prompt.startswith("Enter a package name"):
+        return ("software", "amazon", "foo", RESOURCE.lower())
+    if prompt.startswith("Choose codegen model"):
+        return "2"
+    return ""
+
+
+def test_java_language_plugin_module_is_set():
+    plugin = JavaLanguagePlugin()
+    assert plugin.MODULE_NAME
+
+
+def test_initialize(project):
+    assert (project.root / "README.md").is_file()
+
+    pom_tree = ET.parse(str(project.root / "pom.xml"))
+    namespace = {"maven": "http://maven.apache.org/POM/4.0.0"}
+    actual_group_id = pom_tree.find("maven:groupId", namespace)
+    expected_group_id = "software.amazon.foo.{}".format(RESOURCE.lower())
+    assert actual_group_id.text == expected_group_id
+    path = project.root / "template.yml"
+    with path.open("r", encoding="utf-8") as f:
+        template = yaml.safe_load(f)
+
+    handler_properties = template["Resources"]["TypeFunction"]["Properties"]
+
+    code_uri = "./target/{}-handler-1.0-SNAPSHOT.jar".format(project.hypenated_name)
+    assert handler_properties["CodeUri"] == code_uri
+    handler = "{}.HandlerWrapper::handleRequest".format(expected_group_id)
+    assert handler_properties["Handler"] == handler
+    assert handler_properties["Runtime"] == project._plugin.RUNTIME
+
+
+def test_generate(project):
+    project.load_schema()
+
+    generated_root = project._plugin._get_generated_root(project)
+    generated_tests_root = project._plugin._get_generated_tests_root(project)
+
+    # generated root shouldn't be present
+    assert not generated_root.is_dir()
+    assert not generated_tests_root.is_dir()
+
+    project.generate()
+
+    src_file = generated_root / "test"
+    src_file.touch()
+
+    test_file = generated_tests_root / "test"
+    test_file.touch()
+
+    project.generate()
+
+    # asserts we remove existing files in the tree
+    assert not src_file.is_file()
+    assert not test_file.is_file()
+
+
+def make_target(project, count):
+    target = project.root / "target"
+    target.mkdir(exist_ok=True)
+    jar_paths = []
+    for i in range(count):
+        jar_path = target / "{}-{}.0-SNAPSHOT.jar".format(project.hypenated_name, i)
+        jar_path.touch()
+        jar_paths.append(jar_path)
+    return jar_paths
+
+
+def test__find_jar_zero(project):
```

Comment:
I don't think we need a lot of these tests they are duplicates of what are in: https://github.com/aws-cloudformation/cloudformation-cli-java-plugin/blob/master/tests/test_codegen.py#L94

Nothing in init/generate is newly tested in these.

Reply:
@rjlohan